cmd/ssh.go

```diff
@@ -160,7 +160,12 @@ func ensureHosts(e types.Exec) {
 		ui.Debugf("Failed to remove known_hosts entry: %v", err)
 	}
 
-	if err := ssh.AddHost("uw:"+e.ID, e.Connection.Host, e.Connection.User, e.Connection.Port); err != nil {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		ui.Errorf("Failed to get user home directory: %v", err)
+		os.Exit(1)
+	}
+	if err := ssh.AddHost("uw:"+e.ID, e.Connection.Host, e.Connection.User, e.Connection.Port, filepath.Join(homeDir, ".ssh", "config")); err != nil {
```

Comment:
I think we can defer all ssh related config and settings to the ssh package. This way all the settings are in one place. i.e. you don't need to dependency inject this.

Reply:
In this case we would have to use a global variable if we still want to be able to test the code or remove the tests. Which do you prefer?